packages/taquito-michelson-encoder/src/tokens/comparable/tx_rollup_l2_address.ts

```diff
@@ -0,0 +1,115 @@
+import { BaseTokenSchema } from '../../schema/types';
+import { b58decode, encodePubKey, validateAddress, ValidationResult } from '@taquito/utils';
+import { ComparableToken, SemanticEncoding, Token, TokenFactory, TokenValidationError } from "../token";
+
+export class TxRollupL2AddressValidationError extends TokenValidationError {
+  name = 'TxRollupL2AddressValidationError';
+  constructor(public value: unknown, public token: TxRollupL2AddressToken, message: string) {
+    super(value, token, message)
+  }
+}
+
+export class TxRollupL2AddressToken extends ComparableToken {
+  // should be just l2_address?
+  static prim: 'tx_rollup_l2_address' = 'tx_rollup_l2_address';
+
+  constructor(
+    protected val: { prim: string; args: any[]; annots: any[] },
+    protected idx: number,
+    protected fac: TokenFactory
+  ) {
+    super(val, idx, fac)
+  }
+
+  public ToBigMapKey(val: any) {
+    const decoded = b58decode(val)
+    return {
+      key: { bytes: decoded },
+      type: { prim: 'bytes' }
+    }
+  }
+
+  private isValid(value: any): TxRollupL2AddressValidationError | null {
+    if (validateAddress(value) !== ValidationResult.VALID) {
+      throw new TxRollupL2AddressValidationError(value, this, `tx_rollup_l2_address is not valid: ${value}`)
+    }
+    return null
+  }
+
+  public Encode(args: string[]): any {
+    const val = args.pop();
+    if (!val) {
+      throw new TxRollupL2AddressValidationError(val, this, `arg missing to encode: this -> "${val}"`)
+    }
+    const err = this.isValid(val)
+    if (err) {
+      throw err;
+    }
+    return { string: val }
+  }
+
+  public EncodeObject(val: any, semantic?: SemanticEncoding): any {
+    const err = this.isValid(val);
+    if (err) {
+      throw err;
+    }
+
+    if (semantic && semantic[TxRollupL2AddressToken.prim]) {
+      return semantic[TxRollupL2AddressToken.prim](val)
+    }
+    return { string: val }
+  }
+
+  public Execute(val: {bytes: string; string: string}): string {
+    if (val.string) {
+      return val.string
+    }
+    if (!val.bytes) {
+      throw new TxRollupL2AddressValidationError(val, this, `value cannot be missing string and byte value. must have one ${JSON.stringify(val)}`)
+    }
+    return encodePubKey(val.bytes)
```

Comment:
Can you add a unit test where Execute receives bytes? 
encodePubKey is not supporting this type of value.